astropy/io/fits/hdu/hdulist.py

```diff
@@ -1003,6 +1003,11 @@ def writeto(
             warnings.warn("There is nothing to write.", AstropyUserWarning)
             return
 
+        if fileobj is sys.stdout:
+            # special case stdout for debugging convenience
+            # see https://github.com/astropy/astropy/issues/3427
+            fileobj = fileobj.buffer
+
```

Comment:
Maybe move that block after `update_extend` so we have all fileobj stuff at the same location.